### PR TITLE
TCVP-1435 admin resolution submission

### PR DESCRIPTION
**Pull request by karahanb**
# Description

This PR includes the following proposed change(s):

- [TCVP-1435](https://justice.gov.bc.ca/jira/browse/TCVP-1435)
- Added functionality to update a single JJ Dispute through the Oracle Data Interface API based on unique violation ticket number and the jj dispute data being passed in the body.
- Added feature for changing JJDispute status to 'Review' when updating the JJDispute for admin submission.
- Regenerated OpenAPI spec of OracleDataApi.v1_0.json and added admin resolution submission endpoint to staff api in order to send update JJ Dispute request to oracle data api.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by KDNeufeld — APPROVED** [review]

**Comment by KDNeufeld** [comment]
I'm seeing some null reference warnings on the JJDisputeService.

File: src/backend/TrafficCourts/Staff.Service/Controllers/JJController.cs [hunk 1]
```diff
@@ -96,4 +96,51 @@ public async Task<IActionResult> GetJJDisputeAsync(string JJDisputeId, Cancellat
...
+        {
+            return new HttpError(e.StatusCode, e.Message);
+        }
+        catch (ApiException e) when (e.StatusCode == StatusCodes.Status404NotFound)
+        {
+            return new HttpError(e.StatusCode, e.Message);
+        }
+        catch (ApiException e)
+        {
+            _logger.LogError(e, "Error retrieving Dispute from oracle-data-api");
```
[the remark this answers is not in the document]
Comment (karahanb) [reply src/backend/TrafficCourts/Staff.Service/Controllers/JJController.cs:136]: Yes, that sounds better actually. I copied some parts from other api responses so overlooked some description and error text updates. Will update these error logs.

File: src/backend/TrafficCourts/Staff.Service/Controllers/JJController.cs [hunk 2]
```diff
@@ -96,4 +96,51 @@ public async Task<IActionResult> GetJJDisputeAsync(string JJDisputeId, Cancellat
...
+            return new HttpError(StatusCodes.Status500InternalServerError, e.Message);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error updating Dispute in oracle-data-api");
```
[the remark this answers is not in the document]
Comment (karahanb) [reply src/backend/TrafficCourts/Staff.Service/Controllers/JJController.cs:141]: I agree that mentioning oracle-data-api is irrelevant for this case. So we can just keep it very generic like "General Exception: server error submitting Admin Resolution" for this one I think.

File: src/backend/TrafficCourts/Staff.Service/Controllers/JJController.cs [hunk 3]
```diff
@@ -96,4 +96,51 @@ public async Task<IActionResult> GetJJDisputeAsync(string JJDisputeId, Cancellat
...
+    /// Updates a single JJ Dispute through the Oracle Data Interface API based on unique violation ticket number and the jj dispute data being passed in the body.
+    /// </summary>
+    /// <param name="ticketNumber">Unique identifier for a specific JJ Dispute record.</param>
+    /// <param name="jjDispute"></param>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    /// <response code="200">Admin resolution is submitted. The JJ Dispute is updated.</response>
+    /// <response code="400">The request was not well formed. Check the parameters.</response>
+    /// <response code="401">Unauthenticated.</response>
+    /// <response code="403">Forbidden, wrong user roles.</response>
```
[the remark this answers is not in the document]
Comment (karahanb) [reply src/backend/TrafficCourts/Staff.Service/Controllers/JJController.cs:109]: You're right. 401 and 403 are not thrown, it only throws 400 and 404. I will remove them from the description.

**Comment by karahanb** [comment]
> I'm seeing some null reference warnings on the JJDisputeService (line 43).

You can ignore this null reference warning because it is referring to the BaseUrl property of Oracle Data Api Configuration which is already validated at application settings on startup.

**Pull request merged**